// @commonground/design-system/dist/components/Form/ErrorMessage/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.StyledErrorMessage = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _styledComponents = _interopRequireDefault(require("styled-components"));

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
var getSvg = function getSvg(fillColor) {
  return "<svg viewBox=\"0 0 16 16\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M16 16H0V0h16v16zm-9-5v2h2v-2H7zm0-8v6h2V3H7z\" fill=\"".concat(fillColor, "\"/></svg>");
};

var StyledErrorMessage = /*#__PURE__*/_styledComponents["default"].p.withConfig({
  displayName: "ErrorMessage__StyledErrorMessage",
  componentId: "sc-1gohwmx-0"
})(["background:url(data:image/svg+xml;base64,", ") no-repeat left center;background-size:", ";font-weight:", ";color:", ";margin:", " 0 ", " 0;padding-left:", ";"], function (p) {
  return btoa(getSvg(p.theme.tokens.colorError));
}, function (p) {
  return p.theme.tokens.spacing05;
}, function (p) {
  return p.theme.tokens.fontWeightBold;
}, function (p) {
  return p.theme.tokens.colorError;
}, function (p) {
  return p.theme.tokens.spacing03;
}, function (p) {
  return p.theme.tokens.spacing06;
}, function (p) {
  return p.theme.tokens.spacing06;
});

exports.StyledErrorMessage = StyledErrorMessage;

var ErrorMessage = function ErrorMessage(_ref) {
  var children = _ref.children,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children"]);
  return /*#__PURE__*/_react["default"].createElement(StyledErrorMessage, props, children);
};

ErrorMessage.propTypes = {
  children: _propTypes.node
};
var _default = ErrorMessage;
exports["default"] = _default;
