// @commonground/design-system/dist/components/Form/TextInput/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var _formik = require("formik");

var _ErrorMessage = _interopRequireDefault(require("../ErrorMessage"));

var _index = require("./index.styles");

var _Input = _interopRequireDefault(require("./Input"));

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
var TextInput = function TextInput(_ref) {
  var children = _ref.children,
      showError = _ref.showError,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "showError"]);

  var _useField = (0, _formik.useField)(props),
      _useField2 = (0, _slicedToArray2["default"])(_useField, 2),
      field = _useField2[0],
      meta = _useField2[1];

  var hasError = typeof showError === 'undefined' ? meta.error && meta.touched : showError;
  var disabled = props.disabled;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_index.StyledLabel, {
    disabled: disabled ? true : undefined
  }, children, /*#__PURE__*/_react["default"].createElement(_formik.Field, (0, _extends2["default"])({
    type: "text",
    as: _Input["default"]
  }, field, props, {
    className: hasError ? 'invalid' : null
  }))), hasError ? /*#__PURE__*/_react["default"].createElement(_ErrorMessage["default"], {
    "data-testid": "error-".concat(field.name)
  }, meta.error) : null);
};

TextInput.propTypes = {
  showError: _propTypes.bool,
  children: _propTypes.string.isRequired,
  name: _propTypes.string.isRequired,
  size: (0, _propTypes.oneOf)(['xs', 's', 'm', 'l', 'xl']),
  disabled: _propTypes.bool
};
TextInput.defaultProps = {
  size: 'm',
  disabled: false
};
var _default = TextInput;
exports["default"] = _default;
