// @commonground/design-system/dist/components/Table/index.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledChevron = exports.TrAsLinkLink = exports.TrAsLinkTd = exports.TrAsLink = exports.Td = exports.Th = exports.StyledTable = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactRouterDom = require("react-router-dom");

var _IconChevronRight = _interopRequireDefault(require("./IconChevronRight"));

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
var StyledTable = /*#__PURE__*/_styledComponents["default"].table.withConfig({
  displayName: "indexstyles__StyledTable",
  componentId: "sc-1x7t7ql-0"
})(["border-spacing:unset;width:100%;"]);

exports.StyledTable = StyledTable;

var Th = /*#__PURE__*/_styledComponents["default"].th.withConfig({
  displayName: "indexstyles__Th",
  componentId: "sc-1x7t7ql-1"
})(["text-transform:uppercase;text-align:left;padding:", " ", ";border-bottom:1px solid ", ";font-size:", ";"], function (p) {
  return p.theme.tokens.spacing04;
}, function (p) {
  return p.theme.tokens.spacing03;
}, function (p) {
  return p.theme.colorBorderTable;
}, function (p) {
  return p.theme.tokens.fontSizeSmall;
});

exports.Th = Th;

var Td = /*#__PURE__*/_styledComponents["default"].td.withConfig({
  displayName: "indexstyles__Td",
  componentId: "sc-1x7t7ql-2"
})(["padding:", " ", ";border-bottom:1px solid ", ";"], function (p) {
  return p.theme.tokens.spacing04;
}, function (p) {
  return p.theme.tokens.spacing03;
}, function (p) {
  return p.theme.colorBorderTable;
});

exports.Td = Td;

var TrAsLink = /*#__PURE__*/_styledComponents["default"].tr.withConfig({
  displayName: "indexstyles__TrAsLink",
  componentId: "sc-1x7t7ql-3"
})(["cursor:pointer;&:hover{background-color:", ";}&:focus{outline:2px solid ", ";}"], function (p) {
  return p.theme.colorBackgroundTableHover;
}, function (p) {
  return p.theme.colorBorderTableFocus;
});

exports.TrAsLink = TrAsLink;
var TrAsLinkTd = /*#__PURE__*/(0, _styledComponents["default"])(Td).withConfig({
  displayName: "indexstyles__TrAsLinkTd",
  componentId: "sc-1x7t7ql-4"
})(["width:1px;"]);
exports.TrAsLinkTd = TrAsLinkTd;
var TrAsLinkLink = /*#__PURE__*/(0, _styledComponents["default"])(_reactRouterDom.Link).withConfig({
  displayName: "indexstyles__TrAsLinkLink",
  componentId: "sc-1x7t7ql-5"
})(["display:block;border:none;line-height:100%;"]);
exports.TrAsLinkLink = TrAsLinkLink;
var StyledChevron = /*#__PURE__*/(0, _styledComponents["default"])(_IconChevronRight["default"]).withConfig({
  displayName: "indexstyles__StyledChevron",
  componentId: "sc-1x7t7ql-6"
})(["vertical-align:middle;fill:", ";"], function (p) {
  return p.theme.colorTextLabel;
});
exports.StyledChevron = StyledChevron;
