// @commonground/design-system/dist/components/Table/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _reactRouterDom = require("react-router-dom");

var _index = require("./index.styles");

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
var TableContext = (0, _react.createContext)(false);

var Tr = function Tr(_ref) {
  var children = _ref.children,
      to = _ref.to,
      name = _ref.name,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["children", "to", "name"]);
  var history = (0, _reactRouterDom.useHistory)();
  var withLinks = (0, _react.useContext)(TableContext);

  if (withLinks) {
    var handlePress = function handlePress() {
      return to ? history.push(to) : null;
    };

    return /*#__PURE__*/_react["default"].createElement(_index.TrAsLink, (0, _extends2["default"])({
      onClick: handlePress,
      tabIndex: "0",
      onKeyPress: handlePress
    }, props), children, /*#__PURE__*/_react["default"].createElement(_index.TrAsLinkTd, null, to ? /*#__PURE__*/_react["default"].createElement(_index.TrAsLinkLink, {
      to: to,
      "aria-label": name,
      tabIndex: "-1"
    }, /*#__PURE__*/_react["default"].createElement(_index.StyledChevron, null)) : null));
  }

  return /*#__PURE__*/_react["default"].createElement("tr", props, children);
};

Tr.propTypes = {
  children: _propTypes.node,
  to: _propTypes.string,
  name: _propTypes.string
};

var TrHead = function TrHead(_ref2) {
  var children = _ref2.children,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["children"]);
  var withLinks = (0, _react.useContext)(TableContext);

  if (withLinks) {
    return /*#__PURE__*/_react["default"].createElement("tr", props, children, /*#__PURE__*/_react["default"].createElement(_index.TrAsLinkTd, {
      as: "th"
    }));
  }

  return /*#__PURE__*/_react["default"].createElement("tr", props, children);
};

TrHead.propTypes = {
  children: _propTypes.node
};

var Table = function Table(_ref3) {
  var withLinks = _ref3.withLinks,
      children = _ref3.children,
      props = (0, _objectWithoutProperties2["default"])(_ref3, ["withLinks", "children"]);
  return /*#__PURE__*/_react["default"].createElement(TableContext.Provider, {
    value: withLinks
  }, /*#__PURE__*/_react["default"].createElement(_index.StyledTable, props, children));
};

Table.propTypes = {
  children: _propTypes.node,
  withLinks: _propTypes.bool
};
Table.defaultProps = {
  withLinks: false
};
Table.Th = _index.Th;
Table.Td = _index.Td;
Table.Tr = Tr;
Table.TrHead = TrHead;
var _default = Table;
exports["default"] = _default;
