// @commonground/design-system/dist/components/Collapsible/index.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CollapsibleBody = exports.CollapsibleChevron = exports.CollapsibleTitle = exports.CollapsibleButton = exports.CollapsibleWrapper = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _IconFlippingChevron = _interopRequireDefault(require("../IconFlippingChevron"));

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
// The animation uses 10% of the viewport height as an estimation for the amount of height to animate. When
// the content grows larger that this it will probably fail
var CollapsibleWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  displayName: "indexstyles__CollapsibleWrapper",
  componentId: "sc-1udtzwt-0"
})(["cursor:pointer;.collapsible-enter{max-height:0;}.collapsible-enter-active{max-height:10vh;transition:max-height 300ms;}.collapsible-exit{max-height:10vh;}.collapsible-exit-active{max-height:0;transition:max-height 300ms;}"]);

exports.CollapsibleWrapper = CollapsibleWrapper;

var CollapsibleButton = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  displayName: "indexstyles__CollapsibleButton",
  componentId: "sc-1udtzwt-1"
})(["display:flex;align-items:center;"]);

exports.CollapsibleButton = CollapsibleButton;

var CollapsibleTitle = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  displayName: "indexstyles__CollapsibleTitle",
  componentId: "sc-1udtzwt-2"
})(["flex-grow:1;"]);

exports.CollapsibleTitle = CollapsibleTitle;
var CollapsibleChevron = /*#__PURE__*/(0, _styledComponents["default"])(_IconFlippingChevron["default"]).withConfig({
  displayName: "indexstyles__CollapsibleChevron",
  componentId: "sc-1udtzwt-3"
})(["width:", ";height:", ";flex-grow:0;transition:300ms ease-in-out;"], function (p) {
  return p.theme.tokens.spacing06;
}, function (p) {
  return p.theme.tokens.spacing06;
});
exports.CollapsibleChevron = CollapsibleChevron;

var CollapsibleBody = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  displayName: "indexstyles__CollapsibleBody",
  componentId: "sc-1udtzwt-4"
})(["overflow:hidden;"]);

exports.CollapsibleBody = CollapsibleBody;
