// @commonground/design-system/dist/components/Toaster/index.styles.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CloseButton = exports.AlertWrapper = exports.ToasterRootStyle = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _index = require("../Alert/index.styles");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  #toaster-root {\n    position: fixed;\n    top: ", ";\n    ", "\n    z-index: 1;\n    overflow: hidden;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ToasterRootStyle = /*#__PURE__*/(0, _styledComponents.createGlobalStyle)(_templateObject(), function (p) {
  return p.theme.tokens.spacing07;
}, function (_ref) {
  var theme = _ref.theme,
      position = _ref.position;
  return position === 'topright' ? "right: ".concat(theme.tokens.spacing06, ";") : "left: ".concat(theme.tokens.spacing06, ";");
});
exports.ToasterRootStyle = ToasterRootStyle;

var getStartPosition = function getStartPosition(position) {
  switch (position) {
    case 'topleft':
      return 'translate(-100%, 0)';

    case 'topright':
    default:
      return 'translate(100%, 0)';
  }
};

var AlertWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  displayName: "indexstyles__AlertWrapper",
  componentId: "k45xw3-0"
})(["position:relative;z-index:1;display:flex;width:23rem;margin-bottom:", ";transform:", ";opacity:1;&.show-enter,&.show-enter-done{transition:transform ", "ms ease-out;transform:translate(0,0);}&.hide-enter,&.hide-enter-done{transition:transform ", "ms ease-in,opacity ", "ms ease-in;transform:translate(0,-100%);opacity:0;z-index:0;}& ", "{flex-grow:1;}& ", ",& ", "{margin-right:", ";}"], function (p) {
  return p.theme.tokens.spacing05;
}, function (_ref2) {
  var position = _ref2.position;
  return getStartPosition(position);
}, function (p) {
  return p.transitionInSpeed;
}, function (p) {
  return p.transitionOutSpeed;
}, function (p) {
  return p.transitionOutSpeed;
}, _index.Container, _index.Title, _index.Content, function (p) {
  return p.theme.tokens.spacing06;
});

exports.AlertWrapper = AlertWrapper;

var CloseButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  displayName: "indexstyles__CloseButton",
  componentId: "k45xw3-1"
})(["position:absolute;top:", ";right:", ";display:flex;align-items:center;justify-content:center;width:2.5rem;height:2.5rem;padding:", ";border:0 none;background:transparent;cursor:pointer;"], function (p) {
  return p.theme.tokens.spacing03;
}, function (p) {
  return p.theme.tokens.spacing03;
}, function (p) {
  return p.theme.tokens.spacing03;
});

exports.CloseButton = CloseButton;
