// @commonground/design-system/dist/components/Toaster/ToastAlert.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _reactTransitionGroup = require("react-transition-group");

var _Alert = _interopRequireDefault(require("../Alert"));

var _CloseIcon = _interopRequireDefault(require("./CloseIcon"));

var _index = require("./index.styles");

// Copyright © VNG Realisatie 2020
// Licensed under the EUPL
//
var transitionInSpeed = 150;
var transitionOutSpeed = 100;
var removeDelay = 100;

var ToastAlert = function ToastAlert(_ref) {
  var mapKey = _ref.mapKey,
      removeToast = _ref.removeToast,
      position = _ref.position,
      children = _ref.children,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["mapKey", "removeToast", "position", "children"]);

  var _useState = (0, _react.useState)(''),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      transitionState = _useState2[0],
      setTransitionState = _useState2[1];

  (0, _react.useEffect)(function () {
    setTransitionState('show');
  }, []);

  var handleClick = function handleClick() {
    setTransitionState('hide');
    setTimeout(function () {
      return removeToast(mapKey);
    }, transitionOutSpeed + removeDelay);
  };

  return /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.CSSTransition, {
    "in": transitionState === 'show',
    timeout: transitionInSpeed,
    classNames: "show"
  }, /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.CSSTransition, {
    "in": transitionState === 'hide',
    timeout: transitionOutSpeed,
    classNames: "hide"
  }, /*#__PURE__*/_react["default"].createElement(_index.AlertWrapper, {
    position: position,
    transitionInSpeed: transitionInSpeed,
    transitionOutSpeed: transitionOutSpeed
  }, /*#__PURE__*/_react["default"].createElement(_index.CloseButton, {
    onClick: handleClick,
    role: "button"
  }, /*#__PURE__*/_react["default"].createElement(_CloseIcon["default"], {
    size: 20
  })), /*#__PURE__*/_react["default"].createElement(_Alert["default"], props, children))));
};

ToastAlert.propTypes = {
  mapKey: _propTypes.string.isRequired,
  removeToast: _propTypes.func.isRequired,
  position: _propTypes.string.isRequired,
  title: _propTypes.string,
  children: _propTypes.node,
  variant: (0, _propTypes.oneOf)(['info', 'warning', 'success', 'error'])
};
ToastAlert.defaultProps = {
  variant: 'info'
};
var _default = ToastAlert;
exports["default"] = _default;
